R/glex.R

```diff
@@ -199,15 +198,16 @@ glex.ranger <- function(object, x, max_interaction = NULL, features = NULL, prob
 }
 
 # Function to get all subsets of set
-subsets <- function(x) {
-  if (length(x) == 1) {
-    list(integer(0), x)
-  } else {
-    do.call(c, lapply(0:length(x), combn, x = x, simplify = FALSE))
-  }
+subsets <- function(x, max_interaction = 3) {
+  # if (length(x) == 1) {
+  #   list(integer(0), x)
+  # } else {
+  #   do.call(c, lapply(0:length(x), combn, x = x, simplify = FALSE))
+  # }
+  get_all_subsets_cpp(x, max_interaction)
 }
 
-tree_fun_path_dependent <- function(tree, trees, x, all_S) {
+tree_fun_path_dependent <- function(tree, trees, x, all_S, max_interaction = 999) {
```

Comment:
Maybe a lower default will do as well? Then again, the user doesn't really interact with this function, right? So a `NULL`  might also do to carry superseding values.